department-of-veterans-affairs/caseflow
# Alec/12433 add judge and atty selectors to judgeteam page

Resolves #12433

### Description

- `OrganizationsUsers` page has the ability to recognize and properly render a Judge Team.  
- A Judge Team admin can add and remove the Decision Drafting Attorney role from users on the team.  
- Labels identifying the assigned roles of each user (Judge Team Lead, Decision Drafting Attorney & Admin) as well as descriptions of those roles are displayed for user clarity.

### Acceptance Criteria
#### Frontend
- [ ] JudgeTeam page has the ability to assign and remove Decision Drafting Attorney role to team members
- [ ] Labels representing assigned roles for each user are displayed
- [ ] Changes do not modify appearance of other organizations' pages

#### Backend
- [ ] `Organizations::UsersController` accepts `attorney` params and updates the `DecisionDraftingAttorney` role via the `OrganizationsUser` model
- [ ] Roles are properly retrieved via an update to `administered_user_serializer`
- [ ] Functionality is tied to feature flag

#### Additional Changes
- [ ] Ensure Admin role button properly working via updates to the following models: `make_user_admin` in `organizations_user` and `add_user` in `judge_team` 

BEFORE
<img width="1225" alt="Screen Shot 2020-01-16 at 4 47 55 PM" src="https://user-images.githubusercontent.com/18618189/72568058-3b01c900-3885-11ea-8d3d-4ccd2fee2d5d.png">

AFTER
![judge_team](https://user-images.githubusercontent.com/18618189/72569719-08f26600-3889-11ea-92e4-74623c5f910c.gif)

### Testing Plan
1. Login as Judge BVAAABSHIRE and navigate to the team management page http://localhost:3000/organizations/bvaaabshire/users
2. Ensure the "Enable Decision Drafting" button gives a user the role of "Decision Drafting Attorney"
3. Ensure the "Disable Decision Drafting" button removes "Decision Drafting Attorney" role from user
